pkg/sippyserver/pr_commenting_processor.go

```diff
@@ -555,6 +543,59 @@ func (aw *AnalysisWorker) processRiskAnalysisComment(prPendingComment models.Pul
 	log.Debugf("Comment added to pendingComments: %s/%s/%s", pendingComment.org, pendingComment.repo, pendingComment.sha)
 }
 
+func buildComment(sortedAnalysis RiskAnalysisEntryList, sha string) string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("Job Failure Risk Analysis for sha: %s\n\n| Job Name | Failure Risk |\n|:---|:---|\n", sha))
+
+	for _, value := range sortedAnalysis {
+		tableKey := value.Key
+
+		if value.Value.URL != "" {
+			tableKey = fmt.Sprintf("[%s](%s)", value.Key, value.Value.URL)
+		}
+
+		var riskSb strings.Builder
+		riskSb.WriteString(fmt.Sprintf("<b>%s</b>", value.Value.RiskLevel.Name))
+		for i, t := range value.Value.TestRiskAnalysis {
+			if i > 3 {
+				break
+			}
+			if i > 0 {
+				riskSb.WriteString("<br>---")
+			}
+			riskSb.WriteString(fmt.Sprintf("<br>%s", t.Name))
```

Comment:
Maybe italics for the test name?

Reply:
And do we get too cramped if we use a list?

  * *test name here*
    * This test has passed ...
  * *next test name*
    * etc...

Reply:
Great feedback, It gets funky with lists inside the table so I might have to stick with the <br> but let me play around


Job Failure Risk Analysis for sha: 79d237196d93eb92ed58c66497d8718259264226

| Job Name | Failure Risk |
|:---|:---|
|[pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade/1682395379418533888)|**High**<br>*[sig-api-machinery] disruption/cache-oauth-api connection/reused should be available throughout the test*
- This test has passed 99.88% of 827 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-gcp-ovn-upgrade' 'periodic-ci-openshift-release-master-ci-4.14-e2e-gcp-ovn-upgrade'] in the last 14 days.|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1682395377153609728)|**High**<br>*Cluster upgrade.[sig-apps] job-upgrade*
- This test has passed 100.00% of 37 runs on jobs ['periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-upgrade'] in the last 14 days.|
|[pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6/1682395380257394688)|**Low**<br>*[sig-network][Feature:vlan] should create pingable pods with macvlan interface on an in-container master [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*
- This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br>---<br>*[sig-cli] oc idle [apigroup:apps.openshift.io][apigroup:route.openshift.io][apigroup:project.openshift.io][apigroup:image.openshift.io] by name [Suite:openshift/conformance/parallel]*
- This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br><br>Open Bugs<br>https://issues.redhat.com/browse/OCPBUGS-6586<br>---<br>*[sig-network][Feature:tuning] pod should start with all sysctl on whitelist [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*
- This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br>---<br>*[sig-network][Feature:bond] should create a pod with bond interface [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*
- This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial/1682395377082306560)|**Low**<br>*[sig-storage] PersistentVolumes-local  Stress with local volumes [Serial] should be able to process many pods and reuse local volumes [Suite:openshift/conformance/serial] [Suite:k8s]*
- This test has passed 79.31% of 29 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-single-node-serial'] in the last 14 days.|

Reply:
https://stackoverflow.com/questions/19950648/how-to-write-lists-inside-a-markdown-table

I started going down the html path due to the line breaks.  We can use markdown for the bold and italics but line breaks or lists likely need to be html

Reply:
Job Failure Risk Analysis for sha: 79d237196d93eb92ed58c66497d8718259264226

| Job Name | Failure Risk |
|:---|:---|
|[pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade/1682395379418533888)|**High**<br>*[sig-api-machinery] disruption/cache-oauth-api connection/reused should be available throughout the test*<ul><li>This test has passed 99.88% of 827 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-gcp-ovn-upgrade' 'periodic-ci-openshift-release-master-ci-4.14-e2e-gcp-ovn-upgrade'] in the last 14 days.</li></ul>|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1682395377153609728)|**High**<br>*Cluster upgrade.[sig-apps] job-upgrade*<ul><li>This test has passed 100.00% of 37 runs on jobs ['periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-upgrade'] in the last 14 days.</li></ul>|
|[pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6/1682395380257394688)|**Low**<br>*[sig-network][Feature:vlan] should create pingable pods with macvlan interface on an in-container master [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<ul><li>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.</li></ul><br>---<br>*[sig-cli] oc idle [apigroup:apps.openshift.io][apigroup:route.openshift.io][apigroup:project.openshift.io][apigroup:image.openshift.io] by name [Suite:openshift/conformance/parallel]*<ul><li>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.</li></ul><br><br>Open Bugs<br>https://issues.redhat.com/browse/OCPBUGS-6586<br>---<br>*[sig-network][Feature:tuning] pod should start with all sysctl on whitelist [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<ul><li>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.</li></ul><br>---<br>*[sig-network][Feature:bond] should create a pod with bond interface [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<ul><li>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.</li></ul><br>---<br>Showing 4 of 18 test results|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial/1682395377082306560)|**Low**<br>*[sig-storage] PersistentVolumes-local  Stress with local volumes [Serial] should be able to process many pods and reuse local volumes [Suite:openshift/conformance/serial] [Suite:k8s]*<ul><li>This test has passed 79.31% of 29 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-single-node-serial'] in the last 14 days.</li></ul>|

Reply:
I feel like this is a better view but can go either way

Job Failure Risk Analysis for sha: 79d237196d93eb92ed58c66497d8718259264226

| Job Name | Failure Risk |
|:---|:---|
|[pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-gcp-ovn-upgrade/1682395379418533888)|**High**<br>*[sig-api-machinery] disruption/cache-oauth-api connection/reused should be available throughout the test*<br>This test has passed 99.88% of 827 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-gcp-ovn-upgrade' 'periodic-ci-openshift-release-master-ci-4.14-e2e-gcp-ovn-upgrade'] in the last 14 days.|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1682395377153609728)|**High**<br>*Cluster upgrade.[sig-apps] job-upgrade*<br>This test has passed 100.00% of 37 runs on jobs ['periodic-ci-openshift-release-master-ci-4.14-e2e-aws-ovn-upgrade'] in the last 14 days.|
|[pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-metal-ipi-ovn-ipv6/1682395380257394688)|**Low**<br>*[sig-network][Feature:vlan] should create pingable pods with macvlan interface on an in-container master [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<br>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br>---<br>*[sig-cli] oc idle [apigroup:apps.openshift.io][apigroup:route.openshift.io][apigroup:project.openshift.io][apigroup:image.openshift.io] by name [Suite:openshift/conformance/parallel]*<br>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br><br>Open Bugs<br>https://issues.redhat.com/browse/OCPBUGS-6586<br>---<br>*[sig-network][Feature:tuning] pod should start with all sysctl on whitelist [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<br>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br>---<br>*[sig-network][Feature:bond] should create a pod with bond interface [apigroup:k8s.cni.cncf.io] [Suite:openshift/conformance/parallel]*<br>This test has passed 72.09% of 43 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-metal-ipi-ovn-ipv6'] in the last 14 days.<br>---<br>Showing 4 of 18 test results|
|[pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/28075/pull-ci-openshift-origin-master-e2e-aws-ovn-single-node-serial/1682395377082306560)|**Low**<br>*[sig-storage] PersistentVolumes-local  Stress with local volumes [Serial] should be able to process many pods and reuse local volumes [Suite:openshift/conformance/serial] [Suite:k8s]*<br>This test has passed 79.31% of 29 runs on jobs ['periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-single-node-serial'] in the last 14 days.|